light-client/eth-light-client/src/db.rs

```diff
@@ -0,0 +1,53 @@
+use std::sync::{Arc, Mutex};
+
+use ethers::types::Log;
+use eyre::Result;
+use rusqlite::Connection;
+
+use crate::config::Config;
+
+#[derive(Clone)]
+pub struct DB {
+	conn: Arc<Mutex<Connection>>,
+}
+
+impl DB {
+	pub fn new(config: &Config) -> Result<Self> {
+		let conn = if let Some(path) = config.db_path.clone() {
+			Connection::open(path)?
+		} else {
+			Connection::open_in_memory()?
+		};
+
+		Ok(DB {
+			conn: Arc::new(Mutex::new(conn)),
+		})
+	}
+
+	pub fn create_table(&self) -> Result<usize> {
+		let conn = self.conn.lock().expect("acquire mutex");
+		Ok(conn.execute(
+			"CREATE TABLE IF NOT EXISTS logs (block_number INTEGER, log_index INTEGER, log TEXT)",
+			(),
+		)?)
+	}
+
+	pub fn insert_logs(&self, block_number: u64, log_index: u64, log: &str) -> Result<usize> {
+		let conn = self.conn.lock().expect("acquire mutex");
```

Comment:
Expect is not particularly good at this place, I guess.

Reply:
Usually, `unwrap()` is used in [this case](https://doc.rust-lang.org/stable/std/sync/struct.Mutex.html) to propagate panics when a mutex has been poisoned.
IMHO, `expect()` is better.